server/test/unit/server/webservices/views/test_content.py

```diff
@@ -346,6 +346,68 @@ def test_get_content_types_view(self, mock_factory, mock_resp):
         self.assertTrue(response is mock_resp.return_value)
 
 
+class TestContentUnitSearch(unittest.TestCase):
+    """
+    Tests for ContentUnitSearch view.
+    """
+
+    @mock.patch('pulp.server.webservices.views.content.factory')
+    @mock.patch('pulp.server.webservices.views.content.Criteria')
+    def test_add_repo_memberships_empty(self, mock_crit, mock_factory):
+        """
+        Make sure it doesn't do a search for associations if there are no units found
+        """
+        mock_find = mock_factory.repo_unit_association_query_manager().find_by_criteria
+        ContentUnitSearch()._add_repo_memberships([], 'rpm')
+        self.assertEqual(mock_find.call_count, 0)
+
+    @mock.patch('pulp.server.webservices.views.content.factory')
+    def test_add_repo_memberships_(self, mock_factory):
+        """
+        Ensure that _add_repo_memberships adds a list of repos for the the units.
+        """
+        mock_find = mock_factory.repo_unit_association_query_manager().find_by_criteria
+        mock_find.return_value = [{'repo_id': 'repo1', 'unit_id': 'unit1'}]
+        units = [{'_id': 'unit1'}]
+        ret = ContentUnitSearch()._add_repo_memberships(units, 'rpm')
+        self.assertEqual(mock_find.call_count, 1)
+        self.assertEqual(len(ret), 1)
+        self.assertEqual(ret[0].get('repository_memberships'), ['repo1'])
+
+    @mock.patch('pulp.server.webservices.views.content.ContentUnitSearch._add_repo_memberships')
+    @mock.patch('pulp.server.webservices.views.content._process_content_unit')
+    def test_get_results_without_repos(self, mock_process, mock_add_repo):
+        """
+        Get results without the optional `include_repos`.
+        """
+        content_search = ContentUnitSearch()
+        mock_query = mock.MagicMock()
+        mock_search = mock.MagicMock(return_value=['result_1', 'result_2'])
+        serialized_results = content_search.get_results(mock_query, mock_search, {},
+                                                        type_id='mock_type')
+        mock_process.assert_has_calls([mock.call('result_1', 'mock_type'),
+                                       mock.call('result_2', 'mock_type')])
+        self.assertEqual(len(serialized_results), 2)
```

Comment:
You should assert more than just the length of the results - it should be easy to assert the value as well.